metals/src/main/scala/scala/meta/internal/metals/callHierarchy/CallHierarchyItemInfo.scala

```diff
@@ -0,0 +1,16 @@
+package scala.meta.internal.metals.callHierarchy
+
+/**
+ * Data that is preserved between a call hierarchy prepare and incoming calls or outgoing calls request.
+ *
+ * @param symbols The set of symbols concerned by the request.
+ * @param visited Symbols that are already visited to handle recursive cases.
+ * @param isLocal Do we have to search for symbols in the whole workspace.
+ * @param searchLocal Do we have to search for symbols locally.
+ */
+private[callHierarchy] final case class CallHierarchyItemInfo(
+    symbols: Array[String],
+    visited: Array[String],
+    isLocal: Boolean,
+    searchLocal: Boolean,
```

Comment:
Isn't this two conditions related? Maybe a sealed trait with object would work here,

Reply:
related comment https://github.com/scalameta/metals/pull/4115/files#r958979482